stix2/datastore/filters.py

```diff
@@ -44,6 +44,29 @@ def _check_filter_components(prop, op, value):
     return True
 
 
+def _assemble_filters(filter_arg, filters=[]):
+    """Assemble a list of filters.
+
+    This can be used to allow certain functions to work correctly no matter if
+    the user provides a single filter or a list of them.
+
+    Args:
+        filter_arg (Filter or list): The single Filter or list of Filters to be
+            coerced into a list of Filters.
+        filters (list, optional): A list of Filters to be automatically appended.
+
+    Returns:
+        List of Filters.
+
+    """
+    if isinstance(filter_arg, list):
+        filters.extend(filter_arg)
+    else:
+        filters.append(filter_arg)
+
+    return filters
+
```

Comment:
Every call to _assemble_filters() will append more filters to the same list, and return this ever-growing list.  The `filter` parameter shouldn't be defaulted to a mutable value here.

Reply:
It looks like most, but not all uses of `_assemble_filters()` pass the `filters` argument, but we shouldn't use a default mutable argument in this function. It's not as bad in the other places, since we don't mutate or return the default list argument.